ftrackhq/ftrack-web-widget
# feat: Migrate repository to TypeScript

- [x] The PR title is suitable as a release note

## Changes



- Migrated ftrack-web-widget to TypeScript
- Uses same `tsconfig.json` as `yarn create vite` outputs when choosing a TypeScript project.
- Updates jsdoc comments to tsdoc.


## Test



- Check that it builds
- Check that `yarn lint` outputs typescript errors 
- Check that the config looks good
- Check that it outputs d.ts files and that they are picked up when used in other projects.
